src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -138,6 +139,11 @@ internal sealed class SourceMemberMethodSymbol : SourceMethodSymbol
 
             CheckForBlockAndExpressionBody(
                 syntax.Body, syntax.ExpressionBody, syntax, diagnostics);
+
+            if (this.ReturnsByRefReadonly)
+            {
+                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(this);
```

Comment:
I'll move it to when we read the syntax, and tie it to the syntax check, to make sure there are no circular dependencies.

---
In reply to: [113562534](https://github.com/dotnet/roslyn/pull/18715#discussion_r113562534) [](ancestors = 113562534)